test/dialect/postgresql/test_reflection.py

```diff
@@ -932,6 +932,49 @@ def go():
             ],
         )
 
+    @testing.provide_metadata
+    def test_index_reflection_partial(self):
+        """Reflect the filter defintion on partial indexes
+        """
+
+        metadata = self.metadata
+
+        t1 = Table(
+            "table1",
+            metadata,
+            Column("id", Integer, primary_key=True),
+            Column("name", String(20)),
+            Column("x", Integer),
+        )
+        metadata.create_all()
+        with testing.db.connect().execution_options(autocommit=True) as conn:
+            conn.exec_driver_sql("create index idx1 on table1 ((id || name))")
+            conn.exec_driver_sql(
+                "CREATE INDEX idx2 ON table1 (id) WHERE name = 'test'"
+            )
+            conn.exec_driver_sql(
+                "CREATE INDEX idx3 ON table1 (id) WHERE x >= 5"
+            )
+            with testing.expect_warnings(
+                "Skipped unsupported reflection of "
+                "expression-based index idx1",
+                "Predicate of partial index idx2 ignored during reflection",
```

Comment:
**mike bayer** (zzzeek) wrote:

the "predicate" warning can be removed, because that's what we're reflecting now

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2142